subsys/fs/nvs: nvs_write return missing documentation

Sir-Branch opened the pull request:
- Documentation of the 0 return value for ns_write function
- Ajusted lines length limit from 80 to 100
- Fixed extra and missing parameters for struct nvs_fs
- Misc spelling/grammar changes

Linked to stale issue #34204

Laczen reviewed (approved):
@Sir-Branch, thank you for these changes.

de-nordic commented on include/fs/nvs.h:
```diff
@@ -102,8 +99,9 @@ int nvs_clear(struct nvs_fs *fs);
- * of bytes requested to be written. On error returns -ERRNO code.
+ * @return Number of bytes written. On success, it will be equal to the number of bytes requested
+ * to be written. When a rewrite of the same data already stored is attempted, nothing is written
+ * to flash, thus 0 is returned. On error returns -ERRNO code.
```
We should probably replace the `-ERRNO` with something like `one of negative errno.h defined error codes`, or something like this, to make it more clear.

Sir-Branch committed:
subsys/fs/nvs: nvs_write return missing documentation

- Documentation of the 0 return value for ns_write function
- Ajusted lines length limit from 80 to 100
- Fixed extra and missing parameters for nvs_fs
- Misc spelling/grammar changes

Signed-off-by: Ramiro Merello <rmerello@itba.edu.ar>

de-nordic commented:
@cfriedt Do yo think this could be merged? This is documentation only.
